testing/e2e.sh

```diff
@@ -500,6 +500,28 @@ function test_container_push_all() {
   docker stop -t 0 $cid
 }
 
+function test_container_pull_cache() {
+  cd "${ROOT}"
+  clear_docker
+  scratch_dir="/tmp/_tmp_containnerregistry"
+  cache_dir="$scratch_dir/containnerregistry_cache"
+  bazel_cache="$scratch_dir/bazel_custom_cache"
+
+  # Delete and recreate temp directories.
+  rm -rf $scratch_dir
+  mkdir -p $cache_dir
+  mkdir -p $bazel_cache
+
+  # Run container puller one with caching.
+  DOCKER_REPO_CACHE=$cache_dir PULLER_TIMEOUT=600 bazel --output_base=$bazel_cache test //tests/docker:distoless_fixed_id_digest_test
+
+  # Rerun the puller by changing the puller timeout to force a rerun of of the
+  # target but now using the cache instead of downloading it again.
+  DOCKER_REPO_CACHE=$cache_dir PULLER_TIMEOUT=601 bazel --output_base=$bazel_cache test //tests/docker:distoless_fixed_id_digest_test
```

Comment:
how do we know it worked?

Reply:
As discussed offline there's no easy way to determine this. To ensure the cached pull is faster, we need to setup a full fledged benchmarking that runs several samples and takes an average